### ISSUE TITLE
Cannot resolve deprecated-stylus-loader.js file

### ISSUE THREAD

**Issue by NgDaddy** [nrwl/nx#22250]
### Current Behavior

1. generate app project utilizing `@nx/web` plugin
2. add `style.styl` to the project
3. an error is thrown
`ERROR in <file-path> 4:0-40
Module not found: Error: Can't resolve '<project-path>\node_modules\@nx\webpack\src\plugins\nx-webpack-plugin\lib\webpack\deprecated-stylus-loader.js' in '<project-file-path>'`

### Expected Behavior

1. generate app project utilizing `@nx/web` plugin
2. add `style.styl` to the project
3. the app is bundled with no error related to `*.styl` files

### Steps to Reproduce

1. set NX workspace with 17.3.x version
2. generate app project utilizing `@nx/web` plugin
3. add `style.styl` to the project
4. an error is thrown
`ERROR in <file-path> 4:0-40
Module not found: Error: Can't resolve '<project-path>\node_modules\@nx\webpack\src\plugins\nx-webpack-plugin\lib\webpack\deprecated-stylus-loader.js' in '<project-file-path>'`

### Nx Report

```shell
Node   : 20.10.0
   OS     : win32-x64
   npm    : 10.2.3

   nx                 : 17.3.2
   @nx/js             : 17.3.2
   @nx/jest           : 17.3.2
   @nx/linter         : 17.3.2
   @nx/eslint         : 17.3.2
   @nx/workspace      : 17.3.2
   @nx/angular        : 17.3.2
   @nx/cypress        : 17.3.2
   @nx/devkit         : 17.3.2
   @nx/eslint-plugin  : 17.3.2
   @nx/plugin         : 17.3.2
   @nx/rollup         : 17.3.2
   @nx/storybook      : 17.3.2
   @nrwl/tao          : 17.3.2
   @nx/web            : 17.3.2
   @nx/webpack        : 17.3.2
   typescript         : 5.3.3

   ---------------------------------------

   Community plugins:

   @compodoc/compodoc    : 1.1.23
   @jscutlery/semver     : 4.2.0
   @ngrx/component       : 17.1.1
   @ngrx/component-store : 17.1.1
   @ngrx/effects         : 17.1.1
   @ngrx/entity          : 17.1.1
   @ngrx/operators       : 17.1.1
   @ngrx/signals         : 17.1.1
   @ngrx/store           : 17.1.1
   @nxext/stencil        : 17.2.2
   @storybook/angular    : 7.6.17
   ng-extract-i18n-merge : 2.10.0

   ---------------------------------------

   Local workspace plugins:

         @<scope>/generators
         @<scope>/executors
```

### Failure Logs

```shell
ERROR in <file-path> 4:0-40
Module not found: Error: Can't resolve '<project-path>\node_modules\@nx\webpack\src\plugins\nx-webpack-plugin\lib\webpack\deprecated-stylus-loader.js' in '<project-file-path>'
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by NgDaddy** [comment]
My fix proposal #22249

**Comment by NgDaddy** [comment]
I was on my vacation last week @ndcunningham . But I need it for the version 17th, not 18th. It this possible with `master` branch?
Btw I can see you took over my job.
